dask/bytes/core.py

```diff
@@ -0,0 +1,164 @@
+    """ Given path return dask.delayed file-like objects
+
+    Parameters
+    ----------
+    path: string
+        Filename or globstring
+    compression: string
+        Compression to use.  See ``dask.bytes.compression.files`` for options.
+    **kwargs: dict
+        Options to pass to storage backend.  Often used to pass authentication
+        information to S3 or HDFS.
+
+    Examples
+    --------
+    >>> files = open_files('2015-*-*.csv')  # doctest: +SKIP
+    >>> files = open_files('s3://2015-*-*.csv.gz', compression='gzip')  # doctest: +SKIP
+
+    Returns
+    -------
+    List of ``dask.delayed`` objects that compute to file-like objects
+    """
+    if compression is not None and compression not in compress_files:
+        raise ValueError("Compression type %s not supported" % compression)
+
+    if '://' in path:
+        protocol, path = path.split('://', 1)
+    else:
+        protocol = 'file'
+
+    try:
+        files = _open_files[protocol](path, **kwargs)
+    except KeyError:
+        raise NotImplementedError("Unknown protocol %s://%s" %
+                                  (protocol, path))
+    if compression:
+        decompress = merge(seekable_files, compress_files)[compression]
+        if sys.version_info[0] < 3:
+            files = [delayed(SeekableFile)(file) for file in files]
+        files = [delayed(decompress)(file) for file in files]
+
+    return files
+
+
+def open_text_files(path, encoding=system_encoding, errors='strict',
+        compression=None, **kwargs):
+    """ Given path return dask.delayed file-like objects in text mode
+
+    Parameters
+    ----------
+    path: string
+        Filename or globstring
+    encoding: string
+    errors: string
+    compression: string
+        Compression to use.  See ``dask.bytes.compression.files`` for options.
+    **kwargs: dict
+        Options to pass to storage backend.  Often used to pass authentication
+        information to S3 or HDFS.
+
+    Examples
+    --------
+    >>> files = open_text_files('2015-*-*.csv', encoding='utf-8')  # doctest: +SKIP
+    >>> files = open_text_files('s3://2015-*-*.csv')  # doctest: +SKIP
+
+    Returns
+    -------
+    List of ``dask.delayed`` objects that compute to text file-like objects
+    """
+    if compression is not None and compression not in compress_files:
+        raise ValueError("Compression type %s not supported" % compression)
+
+    original_path = path
+    if '://' in path:
+        protocol, path = path.split('://', 1)
+    else:
+        protocol = 'file'
+
+    if protocol in _open_text_files and compression is None:
+        return _open_text_files[protocol](path, encoding=encoding,
+                                          errors=errors, **kwargs)
+    elif protocol in _open_files:
+        files = open_files(original_path, compression=compression, **kwargs)
+        if sys.version_info[0] < 3:
+            files = [delayed(SeekableFile)(file) for file in files]
```

Comment:
What is the value to moving the check out of the function?  

I find an extra line to be easier to track than an extra redirection.